wear/src/main/java/io/homeassistant/companion/android/tiles/TemplateTile.kt

```diff
@@ -152,4 +163,60 @@ class TemplateTile : TileService() {
             )
             .build()
     }
+
+    private fun parseHtml(renderedText: String): LayoutElementBuilders.Spannable {
+        val renderedSpanned = fromHtml(renderedText, FROM_HTML_MODE_LEGACY)
+        return LayoutElementBuilders.Spannable.Builder().apply {
+            var start = 0
+            var end = 0
+            while (end < renderedSpanned.length) {
+                end = renderedSpanned.nextSpanTransition(end, renderedSpanned.length, CharacterStyle::class.java)
+
+                val fontStyle = LayoutElementBuilders.FontStyle.Builder().apply {
+                    renderedSpanned.getSpans(start, end, CharacterStyle::class.java).forEach {
+                        when (it) {
+                            is AbsoluteSizeSpan -> setSize(
+                                DimensionBuilders.SpProp.Builder()
+                                    .setValue(it.size / applicationContext.resources.displayMetrics.scaledDensity)
+                                    .build()
+                            )
+                            is ForegroundColorSpan -> setColor(
+                                ColorBuilders.ColorProp.Builder()
+                                    .setArgb(it.foregroundColor)
+                                    .build()
+                            )
+                            is RelativeSizeSpan -> {
+                                val defaultSize = 16 // https://developer.android.com/training/wearables/design/typography
+                                setSize(
+                                    DimensionBuilders.SpProp.Builder()
+                                        .setValue(it.sizeChange * defaultSize)
```

Comment:
Would love to do this more smartly, with real relative sizes. Then I won't have to assume the default size and it would allow for nested size tags. Any tips are welcome :)